setup.py

```diff
@@ -256,7 +234,7 @@ def awscrt_ext():
     extra_link_args = os.environ.get('LDFLAGS', '').split()
     extra_objects = []
 
-    libraries = [x.name for x in AWS_LIBS]
+    libraries = [x.name for x in AWS_LIBS if x.name != 'aws-lc']
```

Comment:
is this necessary? if we're not directly linking aws-lc, why are we directly linking s2n?

If we do need this, let's make it a little more data-driven and give a `link` property to each `AwsLib`. So we'd create the lib like `AwsLib('s2n', link=False)`, and then here we'd be like:
libraries = [x.name for x in AWS_LIBS if x.link]

Reply:
Works for me, I think the even better would be to have `self.link=self.name` by default, and then set it to ':libcrypto.a' for aws-lc.

Reply:
I'm confused...
NOW I see that we didn't include "aws-lc" in the list of libraries to link because we want to link to "libcrypto.a" and the original code would have gotten us "libaws-lc.a".

But with the code as it's written here, we don't include anything about libcrypto in the `libraries` list. Is it somehow getting pulled in transitively through s2n and aws-c-cal? Or is it magic that's somehow letting this all pass?

Reply:
figured it out
we were doing some weird stuff with `libraries` in most code-paths
cleaned it up and pushed